common/types.go

```diff
@@ -163,8 +164,77 @@ func (ep *EvictionPolicy) Parse(s string) error {
 
 type LogConfig struct {
 	Level       LogLevel
-	FilePath    string
 	MaxFileSize uint64
 	FileCount   uint64
+	FilePath    string
 	TimeTracker bool
 }
+
+type Block struct {
+	StartIndex int64
+	EndIndex   int64
+	Size       int64
+	Id         string
+	Modified   bool
+}
+
+// list that holds blocks containing ids and corresponding offsets
+type BlockOffsetList struct {
+	BlockList []*Block //blockId to offset mapping
+	Cached    bool     // is it cached?
+}
+
+func (bol BlockOffsetList) binarySearch(offset int64) (bool, int) {
+	lowerBound := 0
+	higherBound := len(bol.BlockList) - 1
+	for lowerBound <= higherBound {
+		middleIndex := (lowerBound + higherBound) / 2
+		// we found the starting block that changes are being applied to
+		if bol.BlockList[middleIndex].EndIndex > offset && bol.BlockList[middleIndex].StartIndex <= offset {
+			return true, middleIndex
+			// if the end index is smaller or equal then we need to increase our lower bound
+		} else if bol.BlockList[middleIndex].EndIndex <= offset {
+			lowerBound = middleIndex + 1
+			// if the start index is larger than the offset we need to decrease our upper bound
+		} else if bol.BlockList[middleIndex].StartIndex > offset {
+			higherBound = middleIndex - 1
+		}
+	}
+	return false, 0
+}
+
+func (bol BlockOffsetList) FindBlocksToModify(offset, length int64) (int, int64, bool, bool) {
```

Comment:
Can we add comment on top for all the return values as to what they mean.

Reply:
done